ui/src/app/model/db-config.ts

```diff
@@ -1,11 +1,11 @@
 export default interface IDbConfig {
-  dbEngine: string
-  isSharded: boolean
-  hostName: string
-  port: string
-  userName: string
-  password: string
-  dbName: string
+  dbEngine: string| null | undefined
+  isSharded: boolean| null | undefined
+  hostName: string | null | undefined
+  port: string | null | undefined
+  userName: string | null | undefined
+  password: string | null | undefined
+  dbName: string | null | undefined
```

Comment:
What are the usecases where they can be `null` or `undefined`? 
(I know this is probably due to how the code is written in the first place but an example of how the `dbEngine` can be null can provide a clue what is written incorrectly..)

Reply:
These are required fields and wouldn't be null but these were added because typescript upgrade required stronger type assertions. Added ! non null assertion operator to overcome this